src/main/java/pl/edu/agh/firecell/engine/algorithm/FirePropagator.java

```diff
@@ -10,6 +10,8 @@
 import static pl.edu.agh.firecell.engine.algorithm.BasicAlgorithm.*;
 import static pl.edu.agh.firecell.model.Material.AIR;
 import static pl.edu.agh.firecell.model.Material.WOOD;
+import static pl.edu.agh.firecell.model.util.NeighbourUtils.Axis.X;
+import static pl.edu.agh.firecell.model.util.NeighbourUtils.Axis.Z;
```

Comment:
Never do import static. Import classes, and use the class name as prefix to imported thing. It is common practise in Java, it make code readable.